osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs

```diff
@@ -212,7 +215,23 @@ private void performRequest()
                 lastResponse = response;
                 getSetsRequest = null;
 
-                SearchFinished?.Invoke(sets);
+                // check if an non-supporter user used supporter-only filters
+                if (!api.LocalUser.Value.IsSupporter && (searchControl.Ranks.Any() || searchControl.Played.Value != SearchPlayed.Any))
```

Comment:
This type of check will also not scale well if there are more supporter-only filters in the future. A simple change that would improve this would involve removing the `Ranks` and `Played` checks from this main `if`, and then doing

```csharp
    if (filters.Any())
    {
        SearchFinished?.Invoke(SearchResult.SupporterOnlyFilter(filters));
        return;
    }
} // end of supporter check block

SearchFinished?.Invoke(SearchResult.ResultsReturned(sets));
```
lower down.